### PR TITLE
About plugin

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to add a new plugin to let user know which Jdaviz version, etc.

<table>
<tr>
<td valign="top">

</td>
<td>

</td>
</tr>
</table>

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-2662)

### CODE REVIEW

File: jdaviz/configs/default/plugins/about/about.py
Comment (pllim) [inline jdaviz/configs/default/plugins/about/about.py:39]: Do we always want to disable PyPI server hit when we're in CI? 🤔

**Comment by pllim** [comment]
I feel like we don't need user-facing doc since this plugin is purely informational and should be pretty self-explanatory.

File: jdaviz/configs/default/plugins/about/about.vue
Comment (kecnry) [inline jdaviz/configs/default/plugins/about/about.vue:28]: this ends up resulting in change log twice, what about the following?
```suggestion
      <j-docs-link link="https://github.com/spacetelescope/jdaviz/blob/main/CHANGES.rst" linktext="Change Log">
        To see list of changes, go to your version section in the

```

Can/should we link directly to the version?  Or maybe we don't even need "go to your version section" at all (if the intention is to see changes _since_ the installed version or if we think it's just self-explanatory)?

File: jdaviz/configs/default/plugins/about/about.vue
Comment (kecnry) [inline jdaviz/configs/default/plugins/about/about.vue:34]: similar to previous comment
```suggestion
      <j-docs-link link="https://jwsthelp.stsci.edu" linktext="JWST Help Desk">
        For further assistance, please contact the

```

**Review by kecnry — APPROVED** [review]
I like it!  We can always add other version information or parse and show the changelog directly as potential future follow-ups.  Just had two small considerations.

File: jdaviz/configs/default/plugins/about/about.vue
Comment (pllim) [reply jdaviz/configs/default/plugins/about/about.vue:28]: Linking directly to version will need some regex magic. It is possible but not something I want to break my brain over for this ticket. Maybe we can start from the logic here if you really want it: https://github.com/scientific-python/action-check-changelogfile/blob/main/rst_parser.py

File: jdaviz/configs/default/plugins/about/about.vue
Comment (pllim) [reply jdaviz/configs/default/plugins/about/about.vue]: Sure. Applied both your suggestions. Thanks!

File: jdaviz/configs/default/plugins/about/about.vue
Comment (kecnry) [reply jdaviz/configs/default/plugins/about/about.vue]: definitely not necessary now if its not trivial (and since there is no API, super easy to iterate on things later)

**Comment by camipacifici** [comment]
Love it!! Great job!
Is it normal that mine complains about 3.10.dev123?
Screenshot 2024-05-23 at 11 19 45 PM

**Comment by pllim** [comment]
> Is it normal that mine complains about 3.10.dev123?

No, but it is also not a problem with the plugin. Whoever released 3.10.0 didn't tag a 3.11.dev. A dev tag is needed when we release from a release branch and not `main`. Though by now, should we just tag a 4.0.dev, @kecnry ?

**Comment by pllim** [comment]
I just tagged a v3.11.dev if you `git pull upstream --tags` and re-install, you should now see a `3.11.dev29+g930125f4`. Please do that and see if the PyPI warning goes away for you. Thanks!

**Comment by camipacifici** [comment]
Does not complain anymore!
I am not a reviewer but you can consider this an approval.

File: jdaviz/configs/default/plugins/about/about.py
Comment (rosteen) [inline jdaviz/configs/default/plugins/about/about.py:43]:
```suggestion
    url = f"https://pypi.org/pypi/{package_name}/json"
```

Not a fan of f-strings?

**Comment by pllim** [comment]
I think @gibsongreen said he was reviewing this. Maybe I should wait for his approval too.

**Review by gibsongreen — APPROVED** [review]
I really like this addition to GUI, nice design and very useful, great work!

Just one consideration, I know the plugin is titled About, and it might be overkill since we have the documentation link elsewhere in the GUI, but I could see first time users finding the link in this plugin beneficial. 

Also, love thehe branch name :relieved:

File: jdaviz/configs/default/plugins/about/about.py
Comment (pllim) [reply jdaviz/configs/default/plugins/about/about.py]: OK accepted your change. Thanks!

**Pull request merged**
